### PR TITLE
[Merged by Bors] - feat(CI): add a log of the size of the oleans

### PR DESCRIPTION
**Pull request by adomani**
Split off from #16020: it only adds a step that computes the sizes of the oleans.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/build.in.yml [hunk 1]
```diff
@@ -183,6 +183,10 @@ jobs:
             lake build --no-build
           fi
 
+      - name: print the sizes of the oleans
```
Comment (mattrobball) [inline .github/build.in.yml:186]: Can we move it before uploading the cache? I imagine we will want to short circuit if the oleans are too big

Comment (adomani) [reply .github/build.in.yml]: Done!

File: .github/build.in.yml [hunk 2]
```diff
@@ -156,6 +156,10 @@ jobs:
           run: |
             bash -o pipefail -c "env LEAN_ABORT_ON_PANIC=1 lake build --wfail -KCI"
 
+      - name: print the sizes of the oleans
+        run: |
+          du .lake/build/lib/Mathlib || echo "This code should be unreachable"
```
Comment (grunweg) [inline .github/build.in.yml:161]: Noting down for myself: this is using Unix paths --- this is fine as this only runs on the builders, which run Ubuntu. (Up to you if you want to leave a comment about this.)

**Review by grunweg — APPROVED** [review]
Thanks, looks good to me. I'll assume the builders have `du` installed.
maintainer merge

**Comment by adomani** [comment]
> Thanks, looks good to me. I'll assume the builders have `du` installed. maintainer merge

I think that they do.  If they do not, we will find out, since we will reach the unreachable code!

File: .github/build.in.yml (hunk 2 above)
Comment (grunweg) [reply .github/build.in.yml:161]: True. The existence of du itself is probably a Unix-ism also.

**Comment by adomani** [comment]
The `||` will not consume the error message: there will already be whatever Ubuntu said when `du` failed.  Should I add something else as well?

**Comment by mattrobball** [comment]
What does it say?

**Comment by mattrobball** [comment]
Nevermind, I got to a computer and those seem reasonable.

**Comment by mattrobball** [comment]
bors merge

**Comment by adomani** [comment]
Matt, I am glad that you could test this, since I would not have known how to test what the message is *without* `du`!

**Comment by adomani** [comment]
Also, a posteriori, `du` is installed on the runners!  :smile:

**Pull request closed without merging**